core/include/sched.h

```diff
@@ -310,32 +310,41 @@ static inline void sched_runqueue_advance(uint8_t prio)
 }
 
 /**
- * @brief   Count the number of threads in a runqueue up to 2
+ * @brief   Tell if the number of threads in a runqueue is 0
  *
- * @param[in]   prio      the priority of the runqueue to get the length from
- * @returns     number of threads in a runqueue up to 2
+ * @param[in]   prio      The priority of the runqueue to get information of
+ * @return      truth value for that information
+ * @warning     This API is not intended for out of tree users.
+ */
+static inline int sched_runqueue_is_empty(uint8_t prio)
+{
+    return clist_is_empty(&sched_runqueues[prio]);
+}
+
+/**
+ * @brief   Tell if the number of threads in a runqueue is 1
  *
+ * @param[in]   prio      The priority of the runqueue to get information of
+ * @return      truth value for that information
  * @warning     This API is not intended for out of tree users.
- *              Breaking API changes will be done without notice and
- *              without deprecation. Consider yourself warned!
+ */
+static inline int sched_runqueue_is_one(uint8_t prio)
+{
+    return clist_exactly_one(&sched_runqueues[prio]);
+}
+
+/**
+ * @brief   Tell if the number of threads in a runqueue greater or equal 2
  *
- * @retval 0    The runqueue is empty
- * @retval 1    There is exactly one thread in the runqueue
- * @retval 2    There are at least two threads in the runqueue
+ * @param[in]   prio      The priority of the runqueue to get information of
+ * @return      truth value for that information
+ * @warning     This API is not intended for out of tree users.
  */
-static inline uint_fast8_t sched_runqueue_len2(uint8_t prio)
+static inline int sched_runqueue_gte_two(uint8_t prio)
```

Comment:
this name is not nice

Reply:
Maybe `int sched_runq_greater_one(uint8_t prio)`?

Reply:
What do you think about shortening all sched_runqueue_fn to sched_runq_fn?